### PR TITLE
Convert REST API page into category

### PR DESCRIPTION
**Pull request by carolstran**
Inspired by #37, it makes more sense to have the link to the REST API reference in the main documentation sidebar so people can 1) know the content exists and 2) more easily access it. 

To enable this, I also converted the REST API page into the a category. The API reference link lives under this category.

Visual preview ✨ 
Screenshot 2023-01-10 at 17 04 57

### CODE REVIEW

File: docs/rest-api/authentication.md [hunk 1]
```diff
@@ -16,7 +12,7 @@ There are two ways to retrieve this token:
 1. [**Authenticate with user credentials**](#authenticate-with-user-credentials): Use the username and password you used while signing up for the [emnify Portal](https://portal.emnify.com/).
```
Comment (techwriter-dev) [inline docs/rest-api/authentication.md:12]: @carolstran I know this is not within the scope of your changes, but I was reading the content and found some English with repetitive word choices that could be changed to sound better. For instance, here we have **use*, **username**, **used** all in the same sentence. Perhaps:

> 1. [**Authenticate with user credentials**](#authenticate-with-user-credentials): Use the username and password you provided while signing up for the [emnify Portal](https://portal.emnify.com/).

File: docs/rest-api/authentication.md [hunk 2]
```diff
@@ -54,7 +50,7 @@ To authenticate all API calls, you can use this `auth_token` as the bearer token
 This `auth_token` is valid for 240 minutes, so you don’t need to retrieve the `auth_token` before every API call.
 Once the `auth_token` expires, you can use the `refresh_token` to retrieve the new `auth_token`.
 
-### Authenticate with an application token
+## Authenticate with an application token
 
 Since you shouldn't store your emnify user credentials on your application server, you can generate an `application_token` via the [emnify Portal](https://portal.emnify.com/) or the API: `/api/v1/application_token`.
 The request body should have a description of the token usually used to indicate who is using the token and can have an `expiry_date` for the token.
```
Comment (techwriter-dev) [inline docs/rest-api/authentication.md:56]: Another English word choice improvement regarding .**.. usually used ...**:
> ... should have a description of the token **normally** used to indicate who is using the token ...

On a side note, the value for `description` in the example below doesn't indicate "who is using the token":
```
"description": "Token with expiry date",
```

> ... and can have an `expiry_date` for the token.

Does this mean `description` is a **required** key in this JSON object and `expiry_date` is an **optional** key?
- **should** have a description
- **can** have an `expiry_date`

Also, why is `expiry_date` monospaced but `description` isn't?

**Review by techwriter-dev — CHANGES_REQUESTED** [review]
The restructuring looks great.

Sorry that I got off scope with the English and formatting suggestions.

File: docs/rest-api/authentication.md (hunk 2 above)
Comment (carolstran) [reply docs/rest-api/authentication.md:56]: Great questions - admittedly, I don't know the answers on hand. But I'd say this falls more into the scope of our future ticket to verify and update the content. Wdyt?

Either way, I swapped "usually" for "normally" 🫡

Comment (techwriter-dev) [reply docs/rest-api/authentication.md]: I'm adding this to my list of content updates that will be needed that I'll add to the backlog.

**Review by techwriter-dev — APPROVED** [review]

**Pull request merged**